contracts/src/v0.8/dev/functions/AuthorizedOriginReceiverUpgradeable.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import "../vendor/openzeppelin-solidity/v.4.8.0/contracts/utils/structs/EnumerableSet.sol";
+import "../vendor/@openzeppelin/contracts-upgradeable/v4.8.1/contracts/proxy/utils/Initializable.sol";
+
+/**
+ * @notice Modified AuthorizedReciever abstract for use on the FunctionsOracle contract to limit usage
+ * @notice Uses tx.origin instead of msg.sender because the client contract sends messages to the Oracle contract
+ */
+
+abstract contract AuthorizedOriginReceiverUpgradeable is Initializable {
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  event AuthorizedSendersChanged(address[] senders, address changedBy);
+  event AuthorizedSendersActive(address account);
+  event AuthorizedSendersDeactive(address account);
+
+  error EmptySendersList();
+  error UnauthorizedSender();
+  error NotAllowedToSetSenders();
+  error AlreadySet();
+
+  bool private _active;
+  EnumerableSet.AddressSet private s_authorizedSenders;
+  address[] private s_authorizedSendersList;
+
+  /**
+   * @dev Initializes the contract in active state.
+   */
+  function __AuthorizedOriginReceiver_initialize(bool active) internal onlyInitializing {
```

Comment:
Note: same contract as `AuthorizedOriginReceiver`, but extends `Initializable` and swaps `constructor` for `initialize` function